### PR TITLE
vipsdisp 3.1.0 (new formula)

### PR DESCRIPTION
**Pull request by jcupitt**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by jcupitt** [comment]
Hello, this is a formula for vipsdisp:

https://github.com/jcupitt/vipsdisp

It's a gtk4 image viewer, just run `vipsdisp my-cat.jpg` to view an image.

It's interesting because:

- it can view very large images (many 10s of GB) without needing much memory
- it supports many scientific and technical image formats, such as those used for medical imaging
- it has a range of built-in visualisation tools (right-click, select View / Display control bar)
- it has good metadata support
- it has "save as", so you can use it for file format conversion
- it can save images in many formats, including DeepZoom
- it supports animation, transparency, many-page images
- ... and some other features, see the README

File: Formula/v/vipsdisp.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  homepage "https://github.com/jcupitt/vipsdisp"
+  url "https://github.com/jcupitt/vipsdisp/releases/download/v3.1.0/vipsdisp-3.1.0.tar.xz"
+  sha256 "5c40e71c9c60232dcbf2e1c389295a4a102a27603bce994dbb2e35ff4f1844db"
+  license "MIT"
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "gtk4"
```
Comment (chenrui333) [inline Formula/v/vipsdisp.rb:12]:
```suggestion
  depends_on "glib"
  depends_on "gtk4"
```

**Review by chenrui333 — APPROVED** [review]

File: Formula/v/vipsdisp.rb (hunk 1 above)
Comment (jcupitt) [reply Formula/v/vipsdisp.rb:12]: I'm happy to make the change of course, but gtk4 depends on glib, so is this necessary?

**Comment by chenrui333** [comment]
let's not do merge commit, you should use rebase if you want, or just leave as it is

File: Formula/v/vipsdisp.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/v/vipsdisp.rb:12]: yeah, we are transitively declaring all the deps

Comment (chenrui333) [reply Formula/v/vipsdisp.rb:13]: you can see in here, https://github.com/Homebrew/homebrew-core/pulls?q=is%3Apr+declare+indirect+deps+with+linkage

Comment (jcupitt) [reply Formula/v/vipsdisp.rb:13]: I'm not sure this is a good idea, there are a lot more transitive deps, perhaps more than 40. Should I list all of them?

**Comment by chenrui333** [comment]
> let's not do merge commit, you should use rebase if you want, or just leave as it is

please dont do merge commit, you can just leave it.

File: Formula/v/vipsdisp.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/v/vipsdisp.rb:13]: it is just one dependency linked in the runtime, that is why only glib got declared

**Comment by chenrui333** [comment]
@jcupitt, thanks a lot for your contribution to Homebrew! :tada: 🥇

**Pull request merged**